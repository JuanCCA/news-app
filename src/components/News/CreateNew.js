import React, {useState} from 'react'
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import FormCreateNew from './FormCreateNew';

const CreateNew = () => {

    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
  

    return (
        <>
          <Button variant="dark" 
                  onClick={handleShow}
          >
            <img 
                src="./plus.png"
                width="25"
                height="25"
                className="d-inline-block align-top"   
                alt=""                 
                >
            </img> 
            Create New
          </Button>
    
          <Modal show={show} onHide={handleClose}>
            <Modal.Header closeButton>
              <Modal.Title>Create a New</Modal.Title>
            </Modal.Header>
            <Modal.Body>

                <FormCreateNew />
                
            </Modal.Body>

          </Modal>
        </>
      );
}

export default CreateNew
