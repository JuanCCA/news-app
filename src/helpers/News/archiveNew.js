import { fetchAnonymous } from "../Petitions/fetch";


export const archiveNew = async (_id) => {

    //const url='http://localhost:4000/api/news/archive_new';
    
    const payload = {
        _id: _id
    }

    
    const resp = await fetchAnonymous('news/archive_new', payload, 'POST')
    const news = await resp.json()
    return news

}