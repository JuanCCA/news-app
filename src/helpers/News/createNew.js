import { fetchAnonymous } from "../Petitions/fetch";


export const createNew = async (title,description,content,author) => {

    //const url='http://localhost:4000/api/news/archive_new';
    
    const payload = {
        title,
        description,
        content,
        author,
    }

    const resp = await fetchAnonymous('news/create_new', payload, 'POST')
    const response = await resp.json()
    return response

}