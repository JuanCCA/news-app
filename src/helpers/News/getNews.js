import { fetchAnonymous } from "../Petitions/fetch";

export const getNews = async () => {

    const resp = await fetchAnonymous( 'news/get_news', null)
    const news = await resp.json()

    return news

}