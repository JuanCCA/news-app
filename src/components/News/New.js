import React from 'react'
import Card from 'react-bootstrap/Card'
import {archiveNew} from '../../helpers/News/archiveNew'
import {formatDate} from '../../helpers/Utils/dateFormater'
import Button from 'react-bootstrap/Button'


const New = ({title,description,date,content,author,archiveDate,_id, setNews}) => {


    const formatedDate = formatDate(date)

    // Archive a New, adding an archiveDate to this object in DB
    const handlerArchive = async() => {
        const resp = await archiveNew(_id)
        if(resp === 200){
            handlerDeleteFromList()
        }
    }

    // Delete this item from the list 
    const handlerDeleteFromList = () => {
        setNews( (news) => {
            console.log(news)
            return news.filter( doc => doc._id !== _id)
        })
    }

    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Date: {formatedDate} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col" >
                    <Button onClick={handlerArchive}
                            variant="dark">
                        <img 
                        src="./eye-off.png"
                        width="25"
                        height="25"
                        className="d-inline-block align-top"   
                        alt=""                 
                        >
                    </img> Archive
                    </Button>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}


export default New
