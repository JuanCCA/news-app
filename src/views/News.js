import React, { useState } from 'react'
import New from '../components/New'
import { useFetchNews } from '../hooks/useFetchNews'

const News = () => {
 
    const [news,setNews] = useState([]) // funcion

    const { data:newsList, loading } = useFetchNews( news );

    return (
        <>
            <p className="display-1 m-auto">Noticias</p>
            <div align="center">
            { loading && <p className="animate__animated animate__flash">Loading</p> }
                <ol>
                    {
                        newsList.map( item => (
                            <New key={item._id} {...item} setNews={setNews} />
                        ))  
                    }
                </ol>
            </div>
            
        </>
    )
}



export default News
