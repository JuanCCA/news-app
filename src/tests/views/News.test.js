import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import  News  from '../../views/News';




describe('Testing <News /> Component', () => {
    

    let wrapper = shallow(<News />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<News />);
    });

    test('should create the component correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });


})

