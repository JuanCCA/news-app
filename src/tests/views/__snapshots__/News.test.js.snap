// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <News /> Component should create the component correctly 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center"
  >
    <p
      className="display-1 m-auto"
    >
      News
    </p>
    <p
      className="m-auto"
    >
      <CreateNew />
    </p>
  </div>
  <div
    align="center"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
    <ol />
  </div>
</Fragment>
`;
